design-tokens/src/utils.ts

```diff
@@ -49,3 +51,68 @@ export const nonComponentTokens: string[] = [
 ];
 
 export const numberToPixel = (value: number): string => `${value}px`;
+
+/**
+ * Ensure variable references are to valid collections. Log errors for any variables that are referencing invalid Figma files.
+ */
+export const verifyReferences = (
+  localTokens: ApiGetLocalVariablesResponse[],
+) => {
+  const invalidVariables: string[] = [];
+  localTokens.forEach(tokens => {
+    Object.keys(tokens.meta.variableCollections).forEach(key => {
+      const collection = tokens.meta.variableCollections[key];
+      if (collection.remote === true) {
+        if (
+          collection.name === 'color' &&
+          collection.key !== process.env['FIGMA_COLOR_COLLECTION_KEY']
+        ) {
+          collection.variableIds.forEach(id =>
+            invalidVariables.push(tokens.meta.variables[id].id),
+          );
+        } else if (
+          collection.name === 'dimension' &&
+          collection.key !== process.env['FIGMA_DIMENSION_COLLECTION_KEY']
+        ) {
+          collection.variableIds.forEach(id =>
+            invalidVariables.push(tokens.meta.variables[id].id),
+          );
+        } else if (
+          collection.name === 'primitives' &&
+          collection.key !== process.env['FIGMA_PRIMITIVE_COLLECTION_KEY']
+        ) {
+          collection.variableIds.forEach(id =>
+            invalidVariables.push(tokens.meta.variables[id].id),
+          );
+        } else if (
+          collection.name === 'global' &&
+          collection.key !== process.env['FIGMA_GLOBAL_COLLECTION_KEY']
+        ) {
+          collection.variableIds.forEach(id =>
+            invalidVariables.push(tokens.meta.variables[id].id),
+          );
+        }
+      }
+    });
+
+    Object.keys(tokens.meta.variables).forEach(key => {
+      const variable = tokens.meta.variables[key];
+      Object.keys(variable.valuesByMode).forEach(j => {
+        const modeValue = variable.valuesByMode[j];
+        if (
+          typeof modeValue === 'object' &&
+          'id' in modeValue &&
+          invalidVariables.includes(modeValue.id)
+        ) {
+          console.error(
+            `🛑 Invalid collection reference for value of: ${variable.name}`,
```

Comment:
The error message screen shot shown in the PR description reads `Invalid {collection.name} collection ...`, but I don't see that reflected here. I think the collection name context is helpful. Intended? Am I missing it?

Reply:
Is there resolution detail we can provide accompanying this error message? e.g. add a corresponding console.log which states `${variable.name} is referencing an incorrect Figma file. Expecting to reference value from ${process.env[collectionKey]}`.

Would something like this be helpful? If I am a dev encountering this error and unfamiliar with the set up, additional specificity helping me towards resolution would be useful.

In my suggested messaging, its recommending specifying the Figma file. Would we be leaking anything sensitive in doing so?

Reply:
The "color" reference in the screenshot was from an earlier stage of the PR where I only was checking color. Once adding other collections, I decided the inclusion of "color"/"dimension" etc. isn't adding value given you really just need the name of the token itself (and would've require more logic to include anyway so I removed).

Addressed your comment with an update to the error message. Having the collection key wouldn't necessarily be helpful, but I've added clarification that the error should be resolved in Figma so a dev with less context would know what to do.